Standards/scs-0123-v1-mandatory-and-supported-IaaS-services.md

```diff
@@ -40,7 +41,7 @@ The following IaaS APIs MUST be present in SCS-compliant IaaS deployments and co
 :::caution
 
 S3 API implementations may differ in certain offered features.
-CSPs must publicly describe, which implementation they use in their deployment.
+CSPs must publicly describe the endpoints of their S3 solutions and which implementations they use in their deployment.
```

Comment:
My understanding is that we allow for a wide variance of S3 solutions and just require that there is a public description of the implementation that is being used, so users roughly know what to expect.
The very first thing I would expect the implementation description to contain would be the endpoint (e.g. "use the swift endpoint from the OpenStack service catalogue with credentials created and stored via `openstack ec credentials create`"), so I added that the description needs to contain this information and thus made it a tiny bit more specific. Having a human-readable description is non-ideal, but we accepted that.

You seem to have read something else into it including this desire, including a requirement that this must then be automatically be tested. The current test case (as described in the standard, I have not seen the code yet) claims to test the object-store (aka swift) endpoint for S3 compatibility, which is useful information and probably covers most of our clouds. We have not defined any other machine-readable way to specify the S3 endpoint yet, which would be a requirement to make it testable. (I wonder whether we should just add it to a certification table or have it as a Gaia-X VC ...) While these are important thoughts for the future, I don't believe these are touched by my proposed change nor do I believe that we need to stop this standards from being stabilized. I'm happy to be convinced otherwise and take back the proposed change.

Reply:
PS: I remember that there was a discussion of requiring an S3 endpoint to be added to the service catalogue. Unconditionally or maybe only if it differs from swift (or no swift is there). If that is meant by

> The endpoints of services MUST be findable through the catalog list of the identity API.

then we need to define what the service name and type of S3 need to be in order to make it findable. (I am not aware of a standard name and type for it. OTC uses `objectstorage` and `object` for it, which seems not the most straight-forward choice to me. `s3` and `object-storage` or `rgw` and `object-storage` would seem more to me. Note that swift uses `swift` and `object-store`.)

I think we can live with the clouds that do NOT use the swift endpoint not having a standardized s3 endpoint yet, as those are the exception. Having them in public documentation is better than nothing ...
It is work to be done before a v2 though, IMVHO.

Reply:
> I'm happy to be convinced otherwise and take back the proposed change.

If that happens, please leave out the erroneous comma. (English has different rules than German here.)

Reply:
@josephineSei Can you comment? I'm a bit out of my depth here

Reply:
We want this stabilized ASAP, so maybe we have to soften it a bit, I don't know.

Reply:
The test, that is implemented works in the following way:
1. if ONLY the clouds.yaml is provided: We assume that s3 is behind the Swift endpoint and everything is configured to be accessible via openstack commands.
2. if a CSP does NOT have a Swift endpoint (I asked this once in an IaaS call, but everyone had s3 behind Swift), the test can be called with arguments for the endpoint and the credentials. 

So right now, every CSP will most likely be compliant. Only in the second case we either need the endpoint and some credentials to test or the CSP MUST run the test locally.